### ISSUE TITLE
Multilingual support

### ISSUE THREAD

**Issue by Vahsify** [Project-Diablo-2/BH#56]
Hello. I saw that translation lines were added to tbl PD2. I translated PD2 into Russian. Would you like to add multilingual support to BH.dll? It was made by Aglophotis in her fork (https://github.com/Aglophotis/BH ). The only noticed problem with this fork is that Windows 7 does not support multi-languages.

**Comment by xijin1337** [comment]
I would also like to ask about multilingual support. I created a translation in Polish, but BH does not display Polish characters except "Ś"

**Comment by xijin1337** [comment]
> Hello. I saw that translation lines were added to tbl PD2. I translated PD2 into Russian. Would you like to add multilingual support to BH.dll? It was made by Aglophotis in her fork (https://github.com/Aglophotis/BH ). The only noticed problem with this fork is that Windows 7 does not support multi-languages.
Do you have a compiled version of BH from Aglophotis? If so, please post.

**Comment by Vahsify** [comment]
> > Hello. I saw that translation lines were added to tbl PD2. I translated PD2 into Russian. Would you like to add multilingual support to BH.dll? It was made by Aglophotis in her fork (https://github.com/Aglophotis/BH ). The only noticed problem with this fork is that Windows 7 does not support multi-languages.
> 
> Do you have a compiled version of BH from Aglophotis? If so, please post.
> 
> 

Hello. I asked Aglophotis to post a release with fixes.

**Comment by Aglophotis** [comment]
> > Hello. I saw that translation lines were added to tbl PD2. I translated PD2 into Russian. Would you like to add multilingual support to BH.dll? It was made by Aglophotis in her fork (https://github.com/Aglophotis/BH ). The only noticed problem with this fork is that Windows 7 does not support multi-languages.
> 
> Do you have a compiled version of BH from Aglophotis? If so, please post.

Hi, i made a release with the latest edits. If you have any questions, you can create a issue in github in my repository or write a to discord.
https://github.com/Aglophotis/BH/releases/tag/0.1.0

**Comment by Aglophotis** [comment]
> The fork created by Aglophotis has changes unrelated to translation (like increasing the max item name size) that will not be included in the main branch for PD2. I also noticed a lot of changes that don't make sense to me and every commit only says "Minor change" with no description of what actually changed. If we were to include these changes, she would need to make a pull request and I would have to review it and modify. Remember also that compiled versions of BH.dll outside of what's provided by the launcher will _not_ work online

If you are really considering the option of implementing multilingual support, then I can bring my changes to a cleaner and better solution than they are now. There are pieces of code that need to be made more universal. In fact, the whole feature was made to just work, since there is no c++ development experience or knowledge of how diablo works from the inside.

**Comment by Canite** [comment]
> > The fork created by Aglophotis has changes unrelated to translation (like increasing the max item name size) that will not be included in the main branch for PD2. I also noticed a lot of changes that don't make sense to me and every commit only says "Minor change" with no description of what actually changed. If we were to include these changes, she would need to make a pull request and I would have to review it and modify. Remember also that compiled versions of BH.dll outside of what's provided by the launcher will _not_ work online
> 
> If you are really considering the option of implementing multilingual support, then I can bring my changes to a cleaner and better solution than they are now. There are pieces of code that need to be made more universal. In fact, the whole feature was made to just work, since there is no c++ development experience or knowledge of how diablo works from the inside.

Yes, that would be nice. From what I saw, your implementation looks good, I'll just have to go through it a bit. But we have been slowly working on adding multilingual support in game and I would like to extend that to BH as well.

**Comment by Aglophotis** [comment]
> > > The fork created by Aglophotis has changes unrelated to translation (like increasing the max item name size) that will not be included in the main branch for PD2. I also noticed a lot of changes that don't make sense to me and every commit only says "Minor change" with no description of what actually changed. If we were to include these changes, she would need to make a pull request and I would have to review it and modify. Remember also that compiled versions of BH.dll outside of what's provided by the launcher will _not_ work online
> > 
> > 
> > If you are really considering the option of implementing multilingual support, then I can bring my changes to a cleaner and better solution than they are now. There are pieces of code that need to be made more universal. In fact, the whole feature was made to just work, since there is no c++ development experience or knowledge of how diablo works from the inside.
> 
> Yes, that would be nice. From what I saw, your implementation looks good, I'll just have to go through it a bit. But we have been slowly working on adding multilingual support in game and I would like to extend that to BH as well.

Well, I'll open a pull request when I finalize the nuances and bring the code to the product state.

**Comment by xijin1337** [comment]
[Aglophotis](https://github.com/Aglophotis)
Best regards, great job!
It would be good if there was a place where you could report errors. I found an error that does not allow the full translation of the BH menu. 

description for `"holdingItem":`  in interaction-tab is impossible to translate
and description from `"lvl99":` in aura-tab duplicates itself to `"rathma":` and `"pvp":
![ss1](https://github.com/user-attachments/assets/14e7a3eb-893f-4628-af0a-dd40db474c2a)
`

**Comment by xijin1337** [comment]
Hello @Aglophotis! 

I would like to ask is there a chance to integrate UTF-8 support and increase the character limit for item descriptions to the latest version of BH (S11)?
